pocs/camera/camera.py

```diff
@@ -634,47 +699,54 @@ def _process_fits(self, file_path, info):
         fits_utils.update_observation_headers(file_path, info)
         return file_path
 
-    def _create_subcomponent(self, subcomponent, sub_name, class_name, base_class):
+    def _create_subcomponent(self, subcomponent, class_name):
         """
         Creates a subcomponent as an attribute of the camera. Can do this from either an instance
         of the appropriate subcomponent class, or from a dictionary of keyword arguments for the
         subcomponent class' constructor.
 
         Args:
-            subcomponent (instance of class_name | dict): the subcomponent object, or the keyword
+            subcomponent (instance of sub_name | dict): the subcomponent object, or the keyword
                 arguments required to create it.
-            sub_name (str): name of the subcomponent, e.g. 'focuser'. Will be used as the attribute
-                name, and must also match the name corresponding POCS submodule for this
-                subcomponent, e.g. `pocs.focuser`
-            class_name (str): name of the subcomponent class, e.g. 'Focuser'
-            base_class (class): the base class for the subcomponent, e.g.
-                `pocs.focuser.AbtractFocuser`, used to check whether subcomponent is an instance.
+            class_name (str): name of the subcomponent class, e.g. 'Focuser'. Lower cased version
+                will be used as the attribute name, and must also match the name of the
+                corresponding POCS submodule for this subcomponent, e.g. `pocs.focuser`.
         """
         if subcomponent:
+            base_module_name = "pocs.{}.{}".format(class_name.casefold(), class_name.casefold())
+            try:
+                base_module = load_module(base_module_name)
+            except AttributeError as err:
+                self.logger.critical("Couldn't import {} base class module {}!".format(
+                    class_name, base_module_name))
+                raise err
+            base_class = getattr(base_module, "Abstract{}".format(class_name))
+
             if isinstance(subcomponent, base_class):
                 self.logger.debug("{} received: {}".format(class_name, subcomponent))
-                setattr(self, sub_name, subcomponent)
-                getattr(self, sub_name).camera = self
+                setattr(self, class_name.casefold(), subcomponent)
+                getattr(self, class_name.casefold()).camera = self
```

Comment:
This assumes the subcomponent has a `camera` attribute already, which is probably a safe assumption, but could use another `setattr` here.

Reply:
Yeah, the subcomponent classes do always have a `camera` attribute (actually a property, I think), though it might be `None`.